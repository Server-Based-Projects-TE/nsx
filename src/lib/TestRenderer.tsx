import type { RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import type { ReactElement } from 'react'
import React from 'react'
import { BrowserRouter } from 'react-router-dom'

const TestRenderer = (ui: ReactElement): RenderResult => {
  const renderResult = render(<BrowserRouter>{ui}</BrowserRouter>)
  return renderResult
}

export default TestRenderer
