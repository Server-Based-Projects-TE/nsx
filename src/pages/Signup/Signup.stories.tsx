import type { ComponentStory } from '@storybook/react'
import React from 'react'

import Create from './'

export default {
  title: 'Pages/Signup',
  component: Create,
}

const Template: ComponentStory<typeof Create> = () => <Create />

export const Default = Template.bind({})
Default.args = {}
