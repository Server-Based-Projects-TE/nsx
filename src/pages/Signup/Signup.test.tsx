import React from 'react'

import TestRenderer from '../../lib/TestRenderer'

import Signup from './'

test('should render Signup', () => {
  const { container } = TestRenderer(<Signup />)
  expect(container).toBeInTheDocument()
})
