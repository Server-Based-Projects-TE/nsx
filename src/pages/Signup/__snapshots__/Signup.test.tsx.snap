// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Signup 1`] = `
<div
  class="flex flex-col justify-between w-screen h-screen"
  data-cy="signup-page-content-root"
>
  <header
    class="w-full border-b border-gray-200"
    data-cy="header"
  >
    <div
      class="sm:flex-row sm:content-center container flex flex-col flex-wrap items-center h-16 gap-2 mx-auto my-4"
    >
      <a
        class="col-auto"
        data-cy="blog-title-top-page-link"
        href="/"
      >
        <h1
          class="sm:text-2xl text-xl font-bold"
        >
          Today I Learned
        </h1>
      </a>
      <p
        class="text-base text-gray-700 justify-self-start sm:pt-2.5 sm:pl-1.5"
      >
        What Today I Learned
        <span
          aria-label="note emoji"
          class="pl-1"
          role="img"
        >
          üìù
        </span>
      </p>
      <div
        class="sm:flex-grow sm:justify-end flex space-x-2"
      >
        <a
          href="/about"
        >
          <div
            class="hover:text-gray-300 text-lg"
          >
            about
          </div>
        </a>
        <a
          href="https://github.com/laststance/nsx"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="inline w-6 h-6 text-gray-500"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <title>
              Github
            </title>
            <path
              clip-rule="evenodd"
              d="M12 2C6.477 2 2 6.484 2 12.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0112 6.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.202 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.943.359.309.678.92.678 1.855 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0022 12.017C22 6.484 17.522 2 12 2z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </div>
    </div>
  </header>
  <main
    class="container mx-auto flex-grow px-4 py-4"
  >
    <h1
      class="mb-3 text-3xl"
    >
      Signup
    </h1>
    <form
      class="w-full max-w-sm"
    >
      <div
        class="md:flex md:items-center mb-6"
      >
        <div
          class="md:w-1/3"
        >
          <label
            class="md:text-right md:mb-0 block pr-4 mb-1 font-bold text-gray-500"
            for="name"
          >
            Name
          </label>
        </div>
        <div
          class="md:w-2/3"
        >
          <div>
            <div
              class="relative mt-1 rounded-md shadow-sm"
            >
              <input
                class="focus:outline-none  focus:bg-white focus:border-purple-500 w-full px-4 py-2 leading-tight text-gray-700 bg-gray-200 border-2 border-gray-200 rounded appearance-none"
                data-cy="name-input"
                name="name"
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="md:flex md:items-center mb-6"
      >
        <div
          class="md:w-1/3"
        >
          <label
            class="md:text-right md:mb-0 block pr-4 mb-1 font-bold text-gray-500"
            for="password"
          >
            Password
          </label>
        </div>
        <div
          class="md:w-2/3"
        >
          <div>
            <div
              class="relative mt-1 rounded-md shadow-sm"
            >
              <input
                class="focus:outline-none  focus:bg-white focus:border-purple-500 w-full px-4 py-2 leading-tight text-gray-700 bg-gray-200 border-2 border-gray-200 rounded appearance-none"
                data-cy="password-input"
                name="password"
                type="password"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="md:flex md:items-center"
      >
        <div
          class="md:w-1/3"
        />
        <div
          class="md:w-2/3"
        >
          <button
            class="flex justify-center items-center shadow rounded disabled:opacity-70 disabled:cursor-not-allowed focus:outline-none font-semibold bg-blue-500 hover:bg-blue-600 text-white py-2 px-6 text-md"
            data-cy="submit-btn"
            type="submit"
          >
            <span
              class="mx-2"
            >
              Submit
            </span>
          </button>
        </div>
      </div>
    </form>
  </main>
  <footer
    class="w-full"
  >
    <div
      class="flex flex-col items-center w-full px-6 border-t border-gray-200"
    >
      <div
        class="sm:w-2/3 py-6 text-center"
      >
        <p
          class="text-sm text-gray-600"
        >
          ¬© 2022 by Ryota Murakami
        </p>
      </div>
    </div>
  </footer>
</div>
`;
